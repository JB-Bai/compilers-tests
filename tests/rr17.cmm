struct O4jjay{
	int kEbwypAnC[0x8b][006106][074215][0xDD], kPl_1r;
}
WbrFbaD(int sGPhSHS, struct{
	struct AqvcjjqzDKP yWcgOo = ucuv0(43.3);
}
gqxZSA6, float orMF716hG, struct smOk2XsvrG{
}
SjFhbN[0x1c072][6950], float ucWuxsr[0xc8E4], float u_q2D){
	float nYrWQC = 0360;
	struct i2HjM{
		struct fpEaYcK LDnOZ73n, fC2fiAj[0355764][07640][07315] = 66.11E+1;
	}
	fH7jYLayM_ = 968.092 * 0xC1b;
	struct _oSYOGz4{
		struct hzWZWjJiV opMQSOZp = ZOh50GL() * 671.98;
		int f0C3Qn8jDc = co_X2k();
		float A3JyIC = 01007;
		float KdfyYPp8 =(! avo_3H()[82.079e+9]);
	}
	XlYCo;
	float BoMloY[001][0xd546] = j9tgwtxAORQ(60741, 24.347E+8) && 335 = tdkucUu() = 052514 * 81106 / 799 - 770.329 = 96.753E5[060][! nlJYBQr] && 033600 * ! LKzo_s0_kSyyM, tdK6WFACRQ[02274];
	struct{
		int Ji2bwdE = 0xEB37 && ya_VQr(dfyRC, gin85, ! 016106 && - 082.369e7 * BvA4sRqX == 006773);
		struct{
			int xAi5FTtaL[022762][0x38D][7828][0x9c] = 073;
			int QHJezb_7 = HsBW() - y03E2N, SUpx6 = 06010, U4sjHdV5D = 939.7259e1 = ! 07151 = 0xB963a && 530.670e+0 <= - !(- 91486) / 41.48e8 <= 160.36e1.ho7lUnv;
		}
		dl1Gr2[0x2A40];
		struct FL98ND DBj9kEFq[0x353Ae];
	}
	iDIiQZ = JNkdXqRO, uTDtaiEJj = - 254777 * 0x9ff + 7998.5950E8 != ! ! ! UmRLg4fbX(wWLBssK).tYy6N - 05453[huM7iUoq - 40.16e-7.pafW <= viiBwqi - kU1F = - -(ryqN8mJdD8 - !(genJNz + DGVlM()) / Pei) = 06462[! 55.88E-2 = gyfvpV ==(d6NqNLtCL0C || Gq4yU)] == 435.66[616.734] * 0x2099 || 679.3481e-3 || 172.13 + 86055];
	! 2716.9996e+1;
}
struct lSYWBWFe G_43CDeL(float Hdk559iSb){
	if(wgVlvYy){
		while(961) HhmDtB;
		return zfjE9slWZ(0xA, HHGa1I(0xbd0A8 != HKYlFJyGG = 002634 || - 399914 || - 0x4CD7 && 182.14E1 / - 1.18 * 0x6e * 2608.509e+6, 0xEA, 53) * 88.281, askLcopA - gF2hBJm(KwgTI0bf));
	}
	else 0x86;
}
float RvGOUZMhd(float CU5a75d7){
	- zoJwC4KQo();
}

