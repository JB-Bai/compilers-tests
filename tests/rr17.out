Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        StructSpecifier (1)
          STRUCT
          OptTag (1)
            ID: O4jjay
          LC
          DefList (2)
            Def (2)
              Specifier (2)
                TYPE: int
              DecList (2)
                Dec (2)
                  VarDec (2)
                    VarDec (2)
                      VarDec (2)
                        VarDec (2)
                          VarDec (2)
                            ID: kEbwypAnC
                          LB
                          INT: 139
                          RB
                        LB
                        INT: 3142
                        RB
                      LB
                      INT: 30861
                      RB
                    LB
                    INT: 221
                    RB
                COMMA
                DecList (2)
                  Dec (2)
                    VarDec (2)
                      ID: kPl_1r
              SEMI
          RC
      FunDec (4)
        ID: WbrFbaD
        LP
        VarList (4)
          ParamDec (4)
            Specifier (4)
              TYPE: int
            VarDec (4)
              ID: sGPhSHS
          COMMA
          VarList (4)
            ParamDec (4)
              Specifier (4)
                StructSpecifier (4)
                  STRUCT
                  LC
                  DefList (5)
                    Def (5)
                      Specifier (5)
                        StructSpecifier (5)
                          STRUCT
                          Tag (5)
                            ID: AqvcjjqzDKP
                      DecList (5)
                        Dec (5)
                          VarDec (5)
                            ID: yWcgOo
                          ASSIGNOP
                          Exp (5)
                            ID: ucuv0
                            LP
                            Args (5)
                              Exp (5)
                                FLOAT: 43.299999
                            RP
                      SEMI
                  RC
              VarDec (7)
                ID: gqxZSA6
            COMMA
            VarList (7)
              ParamDec (7)
                Specifier (7)
                  TYPE: float
                VarDec (7)
                  ID: orMF716hG
              COMMA
              VarList (7)
                ParamDec (7)
                  Specifier (7)
                    StructSpecifier (7)
                      STRUCT
                      OptTag (7)
                        ID: smOk2XsvrG
                      LC
                      RC
                  VarDec (9)
                    VarDec (9)
                      VarDec (9)
                        ID: SjFhbN
                      LB
                      INT: 114802
                      RB
                    LB
                    INT: 6950
                    RB
                COMMA
                VarList (9)
                  ParamDec (9)
                    Specifier (9)
                      TYPE: float
                    VarDec (9)
                      VarDec (9)
                        ID: ucWuxsr
                      LB
                      INT: 51428
                      RB
                  COMMA
                  VarList (9)
                    ParamDec (9)
                      Specifier (9)
                        TYPE: float
                      VarDec (9)
                        ID: u_q2D
        RP
      CompSt (9)
        LC
        DefList (10)
          Def (10)
            Specifier (10)
              TYPE: float
            DecList (10)
              Dec (10)
                VarDec (10)
                  ID: nYrWQC
                ASSIGNOP
                Exp (10)
                  INT: 240
            SEMI
          DefList (11)
            Def (11)
              Specifier (11)
                StructSpecifier (11)
                  STRUCT
                  OptTag (11)
                    ID: i2HjM
                  LC
                  DefList (12)
                    Def (12)
                      Specifier (12)
                        StructSpecifier (12)
                          STRUCT
                          Tag (12)
                            ID: fpEaYcK
                      DecList (12)
                        Dec (12)
                          VarDec (12)
                            ID: LDnOZ73n
                        COMMA
                        DecList (12)
                          Dec (12)
                            VarDec (12)
                              VarDec (12)
                                VarDec (12)
                                  VarDec (12)
                                    ID: fC2fiAj
                                  LB
                                  INT: 121844
                                  RB
                                LB
                                INT: 4000
                                RB
                              LB
                              INT: 3789
                              RB
                            ASSIGNOP
                            Exp (12)
                              FLOAT: 661.099976
                      SEMI
                  RC
              DecList (14)
                Dec (14)
                  VarDec (14)
                    ID: fH7jYLayM_
                  ASSIGNOP
                  Exp (14)
                    Exp (14)
                      FLOAT: 968.091980
                    STAR
                    Exp (14)
                      INT: 3099
              SEMI
            DefList (15)
              Def (15)
                Specifier (15)
                  StructSpecifier (15)
                    STRUCT
                    OptTag (15)
                      ID: _oSYOGz4
                    LC
                    DefList (16)
                      Def (16)
                        Specifier (16)
                          StructSpecifier (16)
                            STRUCT
                            Tag (16)
                              ID: hzWZWjJiV
                        DecList (16)
                          Dec (16)
                            VarDec (16)
                              ID: opMQSOZp
                            ASSIGNOP
                            Exp (16)
                              Exp (16)
                                ID: ZOh50GL
                                LP
                                RP
                              STAR
                              Exp (16)
                                FLOAT: 671.979980
                        SEMI
                      DefList (17)
                        Def (17)
                          Specifier (17)
                            TYPE: int
                          DecList (17)
                            Dec (17)
                              VarDec (17)
                                ID: f0C3Qn8jDc
                              ASSIGNOP
                              Exp (17)
                                ID: co_X2k
                                LP
                                RP
                          SEMI
                        DefList (18)
                          Def (18)
                            Specifier (18)
                              TYPE: float
                            DecList (18)
                              Dec (18)
                                VarDec (18)
                                  ID: A3JyIC
                                ASSIGNOP
                                Exp (18)
                                  INT: 519
                            SEMI
                          DefList (19)
                            Def (19)
                              Specifier (19)
                                TYPE: float
                              DecList (19)
                                Dec (19)
                                  VarDec (19)
                                    ID: KdfyYPp8
                                  ASSIGNOP
                                  Exp (19)
                                    LP
                                    Exp (19)
                                      NOT
                                      Exp (19)
                                        Exp (19)
                                          ID: avo_3H
                                          LP
                                          RP
                                        LB
                                        Exp (19)
                                          FLOAT: 82078998528.000000
                                        RB
                                    RP
                              SEMI
                    RC
                DecList (21)
                  Dec (21)
                    VarDec (21)
                      ID: XlYCo
                SEMI
              DefList (22)
                Def (22)
                  Specifier (22)
                    TYPE: float
                  DecList (22)
                    Dec (22)
                      VarDec (22)
                        VarDec (22)
                          VarDec (22)
                            ID: BoMloY
                          LB
                          INT: 1
                          RB
                        LB
                        INT: 54598
                        RB
                      ASSIGNOP
                      Exp (22)
                        Exp (22)
                          Exp (22)
                            ID: j9tgwtxAORQ
                            LP
                            Args (22)
                              Exp (22)
                                INT: 60741
                              COMMA
                              Args (22)
                                Exp (22)
                                  FLOAT: 2434700032.000000
                            RP
                          AND
                          Exp (22)
                            INT: 335
                        ASSIGNOP
                        Exp (22)
                          Exp (22)
                            ID: tdkucUu
                            LP
                            RP
                          ASSIGNOP
                          Exp (22)
                            Exp (22)
                              Exp (22)
                                Exp (22)
                                  Exp (22)
                                    INT: 21836
                                  STAR
                                  Exp (22)
                                    INT: 81106
                                DIV
                                Exp (22)
                                  INT: 799
                              MINUS
                              Exp (22)
                                FLOAT: 770.328979
                            ASSIGNOP
                            Exp (22)
                              Exp (22)
                                Exp (22)
                                  Exp (22)
                                    FLOAT: 9675300.000000
                                  LB
                                  Exp (22)
                                    INT: 48
                                  RB
                                LB
                                Exp (22)
                                  NOT
                                  Exp (22)
                                    ID: nlJYBQr
                                RB
                              AND
                              Exp (22)
                                Exp (22)
                                  INT: 14208
                                STAR
                                Exp (22)
                                  NOT
                                  Exp (22)
                                    ID: LKzo_s0_kSyyM
                    COMMA
                    DecList (22)
                      Dec (22)
                        VarDec (22)
                          VarDec (22)
                            ID: tdK6WFACRQ
                          LB
                          INT: 1212
                          RB
                  SEMI
                DefList (23)
                  Def (23)
                    Specifier (23)
                      StructSpecifier (23)
                        STRUCT
                        LC
                        DefList (24)
                          Def (24)
                            Specifier (24)
                              TYPE: int
                            DecList (24)
                              Dec (24)
                                VarDec (24)
                                  ID: Ji2bwdE
                                ASSIGNOP
                                Exp (24)
                                  Exp (24)
                                    INT: 60215
                                  AND
                                  Exp (24)
                                    ID: ya_VQr
                                    LP
                                    Args (24)
                                      Exp (24)
                                        ID: dfyRC
                                      COMMA
                                      Args (24)
                                        Exp (24)
                                          ID: gin85
                                        COMMA
                                        Args (24)
                                          Exp (24)
                                            Exp (24)
                                              NOT
                                              Exp (24)
                                                INT: 7238
                                            AND
                                            Exp (24)
                                              Exp (24)
                                                Exp (24)
                                                  MINUS
                                                  Exp (24)
                                                    FLOAT: 823689984.000000
                                                STAR
                                                Exp (24)
                                                  ID: BvA4sRqX
                                              RELOP
                                              Exp (24)
                                                INT: 3579
                                    RP
                            SEMI
                          DefList (25)
                            Def (25)
                              Specifier (25)
                                StructSpecifier (25)
                                  STRUCT
                                  LC
                                  DefList (26)
                                    Def (26)
                                      Specifier (26)
                                        TYPE: int
                                      DecList (26)
                                        Dec (26)
                                          VarDec (26)
                                            VarDec (26)
                                              VarDec (26)
                                                VarDec (26)
                                                  VarDec (26)
                                                    ID: xAi5FTtaL
                                                  LB
                                                  INT: 9714
                                                  RB
                                                LB
                                                INT: 909
                                                RB
                                              LB
                                              INT: 7828
                                              RB
                                            LB
                                            INT: 156
                                            RB
                                          ASSIGNOP
                                          Exp (26)
                                            INT: 59
                                      SEMI
                                    DefList (27)
                                      Def (27)
                                        Specifier (27)
                                          TYPE: int
                                        DecList (27)
                                          Dec (27)
                                            VarDec (27)
                                              ID: QHJezb_7
                                            ASSIGNOP
                                            Exp (27)
                                              Exp (27)
                                                ID: HsBW
                                                LP
                                                RP
                                              MINUS
                                              Exp (27)
                                                ID: y03E2N
                                          COMMA
                                          DecList (27)
                                            Dec (27)
                                              VarDec (27)
                                                ID: SUpx6
                                              ASSIGNOP
                                              Exp (27)
                                                INT: 3080
                                            COMMA
                                            DecList (27)
                                              Dec (27)
                                                VarDec (27)
                                                  ID: U4sjHdV5D
                                                ASSIGNOP
                                                Exp (27)
                                                  Exp (27)
                                                    FLOAT: 9397.258789
                                                  ASSIGNOP
                                                  Exp (27)
                                                    Exp (27)
                                                      NOT
                                                      Exp (27)
                                                        INT: 3689
                                                    ASSIGNOP
                                                    Exp (27)
                                                      Exp (27)
                                                        INT: 759354
                                                      AND
                                                      Exp (27)
                                                        Exp (27)
                                                          Exp (27)
                                                            FLOAT: 530.669983
                                                          RELOP
                                                          Exp (27)
                                                            Exp (27)
                                                              MINUS
                                                              Exp (27)
                                                                NOT
                                                                Exp (27)
                                                                  LP
                                                                  Exp (27)
                                                                    MINUS
                                                                    Exp (27)
                                                                      INT: 91486
                                                                  RP
                                                            DIV
                                                            Exp (27)
                                                              FLOAT: 4148000000.000000
                                                        RELOP
                                                        Exp (27)
                                                          Exp (27)
                                                            FLOAT: 1603.599976
                                                          DOT
                                                          ID: ho7lUnv
                                        SEMI
                                  RC
                              DecList (29)
                                Dec (29)
                                  VarDec (29)
                                    VarDec (29)
                                      ID: dl1Gr2
                                    LB
                                    INT: 10816
                                    RB
                              SEMI
                            DefList (30)
                              Def (30)
                                Specifier (30)
                                  StructSpecifier (30)
                                    STRUCT
                                    Tag (30)
                                      ID: FL98ND
                                DecList (30)
                                  Dec (30)
                                    VarDec (30)
                                      VarDec (30)
                                        ID: DBj9kEFq
                                      LB
                                      INT: 218030
                                      RB
                                SEMI
                        RC
                    DecList (32)
                      Dec (32)
                        VarDec (32)
                          ID: iDIiQZ
                        ASSIGNOP
                        Exp (32)
                          ID: JNkdXqRO
                      COMMA
                      DecList (32)
                        Dec (32)
                          VarDec (32)
                            ID: uTDtaiEJj
                          ASSIGNOP
                          Exp (32)
                            Exp (32)
                              Exp (32)
                                Exp (32)
                                  MINUS
                                  Exp (32)
                                    INT: 254777
                                STAR
                                Exp (32)
                                  INT: 2559
                              PLUS
                              Exp (32)
                                FLOAT: 799859474432.000000
                            RELOP
                            Exp (32)
                              Exp (32)
                                NOT
                                Exp (32)
                                  NOT
                                  Exp (32)
                                    NOT
                                    Exp (32)
                                      Exp (32)
                                        ID: UmRLg4fbX
                                        LP
                                        Args (32)
                                          Exp (32)
                                            ID: wWLBssK
                                        RP
                                      DOT
                                      ID: tYy6N
                              MINUS
                              Exp (32)
                                Exp (32)
                                  INT: 2859
                                LB
                                Exp (32)
                                  Exp (32)
                                    Exp (32)
                                      Exp (32)
                                        ID: huM7iUoq
                                      MINUS
                                      Exp (32)
                                        Exp (32)
                                          FLOAT: 0.000004
                                        DOT
                                        ID: pafW
                                    RELOP
                                    Exp (32)
                                      Exp (32)
                                        ID: viiBwqi
                                      MINUS
                                      Exp (32)
                                        ID: kU1F
                                  ASSIGNOP
                                  Exp (32)
                                    Exp (32)
                                      MINUS
                                      Exp (32)
                                        MINUS
                                        Exp (32)
                                          LP
                                          Exp (32)
                                            Exp (32)
                                              ID: ryqN8mJdD8
                                            MINUS
                                            Exp (32)
                                              Exp (32)
                                                NOT
                                                Exp (32)
                                                  LP
                                                  Exp (32)
                                                    Exp (32)
                                                      ID: genJNz
                                                    PLUS
                                                    Exp (32)
                                                      ID: DGVlM
                                                      LP
                                                      RP
                                                  RP
                                              DIV
                                              Exp (32)
                                                ID: Pei
                                          RP
                                    ASSIGNOP
                                    Exp (32)
                                      Exp (32)
                                        Exp (32)
                                          Exp (32)
                                            Exp (32)
                                              INT: 3378
                                            LB
                                            Exp (32)
                                              Exp (32)
                                                NOT
                                                Exp (32)
                                                  FLOAT: 0.558800
                                              ASSIGNOP
                                              Exp (32)
                                                Exp (32)
                                                  ID: gyfvpV
                                                RELOP
                                                Exp (32)
                                                  LP
                                                  Exp (32)
                                                    Exp (32)
                                                      ID: d6NqNLtCL0C
                                                    OR
                                                    Exp (32)
                                                      ID: Gq4yU
                                                  RP
                                            RB
                                          RELOP
                                          Exp (32)
                                            Exp (32)
                                              Exp (32)
                                                FLOAT: 435.660004
                                              LB
                                              Exp (32)
                                                FLOAT: 616.734009
                                              RB
                                            STAR
                                            Exp (32)
                                              INT: 8345
                                        OR
                                        Exp (32)
                                          FLOAT: 0.679348
                                      OR
                                      Exp (32)
                                        Exp (32)
                                          FLOAT: 172.130005
                                        PLUS
                                        Exp (32)
                                          INT: 86055
                                RB
                    SEMI
        StmtList (33)
          Stmt (33)
            Exp (33)
              NOT
              Exp (33)
                FLOAT: 27169.996094
            SEMI
        RC
    ExtDefList (35)
      ExtDef (35)
        Specifier (35)
          StructSpecifier (35)
            STRUCT
            Tag (35)
              ID: lSYWBWFe
        FunDec (35)
          ID: G_43CDeL
          LP
          VarList (35)
            ParamDec (35)
              Specifier (35)
                TYPE: float
              VarDec (35)
                ID: Hdk559iSb
          RP
        CompSt (35)
          LC
          StmtList (36)
            Stmt (36)
              IF
              LP
              Exp (36)
                ID: wgVlvYy
              RP
              Stmt (36)
                CompSt (36)
                  LC
                  StmtList (37)
                    Stmt (37)
                      WHILE
                      LP
                      Exp (37)
                        INT: 961
                      RP
                      Stmt (37)
                        Exp (37)
                          ID: HhmDtB
                        SEMI
                    StmtList (38)
                      Stmt (38)
                        RETURN
                        Exp (38)
                          ID: zfjE9slWZ
                          LP
                          Args (38)
                            Exp (38)
                              INT: 10
                            COMMA
                            Args (38)
                              Exp (38)
                                Exp (38)
                                  ID: HHGa1I
                                  LP
                                  Args (38)
                                    Exp (38)
                                      Exp (38)
                                        Exp (38)
                                          INT: 774312
                                        RELOP
                                        Exp (38)
                                          ID: HKYlFJyGG
                                      ASSIGNOP
                                      Exp (38)
                                        Exp (38)
                                          Exp (38)
                                            INT: 1436
                                          OR
                                          Exp (38)
                                            MINUS
                                            Exp (38)
                                              INT: 399914
                                        OR
                                        Exp (38)
                                          Exp (38)
                                            MINUS
                                            Exp (38)
                                              INT: 19671
                                          AND
                                          Exp (38)
                                            Exp (38)
                                              Exp (38)
                                                Exp (38)
                                                  FLOAT: 1821.400024
                                                DIV
                                                Exp (38)
                                                  MINUS
                                                  Exp (38)
                                                    FLOAT: 1.180000
                                              STAR
                                              Exp (38)
                                                INT: 110
                                            STAR
                                            Exp (38)
                                              FLOAT: 2608508928.000000
                                    COMMA
                                    Args (38)
                                      Exp (38)
                                        INT: 234
                                      COMMA
                                      Args (38)
                                        Exp (38)
                                          INT: 53
                                  RP
                                STAR
                                Exp (38)
                                  FLOAT: 88.280998
                              COMMA
                              Args (38)
                                Exp (38)
                                  Exp (38)
                                    ID: askLcopA
                                  MINUS
                                  Exp (38)
                                    ID: gF2hBJm
                                    LP
                                    Args (38)
                                      Exp (38)
                                        ID: KwgTI0bf
                                    RP
                          RP
                        SEMI
                  RC
              ELSE
              Stmt (40)
                Exp (40)
                  INT: 134
                SEMI
          RC
      ExtDefList (42)
        ExtDef (42)
          Specifier (42)
            TYPE: float
          FunDec (42)
            ID: RvGOUZMhd
            LP
            VarList (42)
              ParamDec (42)
                Specifier (42)
                  TYPE: float
                VarDec (42)
                  ID: CU5a75d7
            RP
          CompSt (42)
            LC
            StmtList (43)
              Stmt (43)
                Exp (43)
                  MINUS
                  Exp (43)
                    ID: zoJwC4KQo
                    LP
                    RP
                SEMI
            RC
